src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/EligibilityService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.confirmationstatementapi.ConfirmationStatementApiApplication;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+import uk.gov.companieshouse.confirmationstatementapi.model.response.EligibilityFailureResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class EligibilityService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfirmationStatementApiApplication.APP_NAME);
+    private final List<EligibilityRule<CompanyProfileApi>> eligibilityRules;
+
+    @Autowired
+    public EligibilityService(@Qualifier("confirmation-statement-eligibility-rules")
+                                                    List<EligibilityRule<CompanyProfileApi>> eligibilityRules){
+        this.eligibilityRules = eligibilityRules;
+    }
+
+    public Optional<EligibilityFailureResponse> checkCompanyEligibility(CompanyProfileApi companyProfile) {
+        return checkCompanyEligibility(companyProfile, null);
+    }
+
+    public Optional<EligibilityFailureResponse> checkCompanyEligibility(CompanyProfileApi companyProfile, Transaction transaction) {
+        try {
+            for (EligibilityRule<CompanyProfileApi> eligibilityRule : eligibilityRules) {
+                eligibilityRule.validate(companyProfile);
+            }
+        } catch (EligibilityException e) {
+            String companyNumber = (transaction != null)?  transaction.getCompanyNumber() : companyProfile.getCompanyNumber();
```

Comment:
not sure its worth passing the transaction in just to get the company number, as we have that in the profile anyway.

I would just stick to `checkCompanyEligibility(profile)` as the only method signature